controllers/consumer_controller.go

```diff
@@ -160,13 +162,21 @@ func (r *ConsumerReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		"toEstimate", len(co.toEstimateInstances),
 	)
 
+	if cmp.Equal(curCons.Status, consumer.Status) {
```

Comment:
looks like the only thing that could change after creating of operator is `consumer.Status.ObservedGeneration`
I guess it should be simpler to compare only generation. 
Also it is unclear why consumer was changed after operator was created? You'd better compare with `co.consumer`

Reply:
in `syncDeploys` we update status
```
	status := &co.consumer.Status
	status.Running = helpers.Ptr2Int32(int32(len(co.runningIds)))
	status.Paused = helpers.Ptr2Int32(int32(len(co.pausedIds)))
	status.Lagging = helpers.Ptr2Int32(int32(len(co.laggingIds)))
	status.Outdated = helpers.Ptr2Int32(int32(len(co.toUpdateInstances)))
	status.Expected = co.consumer.Spec.NumPartitions
	status.Missing = helpers.Ptr2Int32(int32(len(co.missingIds)))
```